vertx-grpc-common/src/main/java/io/vertx/grpc/common/impl/Utils.java

```diff
@@ -16,21 +16,24 @@
 import io.vertx.core.MultiMap;
 
 import java.nio.charset.StandardCharsets;
+import java.util.List;
 import java.util.Map;
 
 public class Utils {
 
   public static void writeMetadata(Metadata metadata, MultiMap mmap) {
     byte[][] t = InternalMetadata.serialize(metadata);
-    for (int i = 0;i < t.length;i+=2) {
+    for (int i = 0; i < t.length; i += 2) {
       mmap.add(new AsciiString(t[i], false), new AsciiString(t[i + 1], false));
     }
   }
 
   public static Metadata readMetadata(MultiMap headers) {
-    byte[][] abc = new byte[headers.size() * 2][];
+    List<Map.Entry<String, String>> entries = headers.entries();
+    // We cannot use headers.size as it's the number of keys, but in a multiple a key can have multiple values.
```

Comment:
Hmmm ... not exactly sure what you're trying to explain here?
I mean I can guess, but a better wording would make it easier.